### ISSUE TITLE
Invalid  YAML parsing of ENTRYPOINT and COMMAND for docker-compose.yml

### ISSUE THREAD

**Issue by markriggins** [aws/amazon-ecs-cli#108]
In a docker-compose.yml we should be able to specify an entrypoint and command with options. The following yaml works for `docker-compose up`, but fails for `ecs-cli compose service up`:
```

ls:
  image: centos:centos7
  entrypoint: ls
  command: -l -a Docker*

```
But fails for ecs-cli with the error: `ls: cannot access l -a Docker*: No such file or directory`
Notice that it is eating the leading dash, and then treating the rest `l -a Docker*` as a SINGLE argument.  Since there is no file name `l -a Docker*` it fails.

Trying work around it:
```
ls:
  image: centos:centos7
  entrypoint: ls
  command: [-l, -a, Docker*]
```
Give the error: `ls: cannot access [-l, -a, Docker*]: No such file or directory`

Trying another work around:
```
ls:
  image: centos:centos7
  entrypoint: ls
  command: [
    "-l", 
    "-a", 
    "Docker*" ]
```
Give the even MORE comical `ls: cannot access "Docker*" ] "-a",  "-l",  [: No such file or directory`

Why does  `ecs-cli` insists on parsing yaml differently than `docker-compose`?  

It would be nice if you could have forked the docker-compose source code, and added AWS features.  If Amazon `insists` on writing it's own code.  Then it should `at least` be compatible.

**Comment by markriggins** [comment]
Final attempt at a work-around:
```
ls:
  image: centos:centos7
  entrypoint: ls -l -a Docker*
```
Give the even worse error: `exec: "ls -l -a Docker*": executable file not found in $PATH`, treating the entire entrypoint (options and all) as a file name.

And the ever-hopeful attempt:
```
ls:
  image: centos:centos7
  entrypoint: [
    ls, 
    '-l',
    '-a', 
    'Docker*' ]
```
Give us `exec: "'Docker*' ] '-a',  '-l', ls,  [": executable file not found in $PATH`

**Comment by markriggins** [comment]
I'll admit it -- I'm persistent if nothing else:

```
ls:
  image: centos:centos7
  entrypoint: 
    - ls
    - '-l'
    - '-a'
    - 'Docker*'
```
Shows some promise but gives us `ls: cannot access '-l': No such file or directory`
But maybe we're onto something finally.

MAYBE just MAYBE one of the dozen ways forms of yaml that `docker-compose` accepts will be accepted by this wonderful ecs-cli command. 

```
ls:
  image: centos:centos7
  entrypoint: 
    - ls
    - -l
    - -a
    - Docker*
```
Surely we `FINALLY` have a winner.  But no, ecs-cli refuses to parse the are only hope of actually passing arguments to the command. 

 `ERRO[0000] Unable to convert yaml to service configs     error=Error parsing: node is not of type string [l]`

**Comment by markriggins** [comment]
# ecs-cli --version
ecs-cli version 0.3.0 (2297ef0)

**Comment by markriggins** [comment]
Our only work around was to create a entrypoint.sh file with: #!/usr/bin/env bash

**Comment by chris-lu** [comment]
Subscribing, facing the same issue.

**Comment by uttarasridhar** [comment]
This is addressed in #121 and released in [v0.4.0](https://github.com/aws/amazon-ecs-cli/releases/tag/v0.4.0).

Trying it out now creates the entry point and command as per expectation. Let us know if you run into issues.
                "entryPoint": [
                    "ls"
                ], 
                "command": [
                    "-l", 
                    "-a", 
                    "Docker*"
                ],

**Comment by markriggins** [comment]
Thankyou !!
